File: check.py
import difflib


def Similarity_Check(answer, input):
    answer_bytes = bytes(answer, 'utf-8')
    input_bytes = bytes(input, 'utf-8')
    answer_bytes_list = list(answer_bytes)
    input_bytes_list = list(input_bytes)

    sm = difflib.SequenceMatcher(None, answer_bytes_list, input_bytes_list)
    similar = sm.ratio()
    return similar

print(Similarity_Check('안녕하세요','안녕하새요'))

File: check.js
const toBytes = (text) => {
  const surrogate = encodeURIComponent(text);
  const result = [];
  for (let i = 0; i < surrogate.length; ) {
    const character = surrogate[i];
    i += 1;
    if (character == "%") {
      const hex = surrogate.substring(i, (i += 2));
      if (hex) {
        result.push(parseInt(hex, 16));
      }
    } else {
      result.push(character.charCodeAt(0));
    }
  }
  return result;
};

const similarity_check = (answer, input) => {
  var difflib = require("difflib");
  var answer_bytes = toBytes(answer);
  var input_bytes = toBytes(input);
  var similarity = new difflib.SequenceMatcher(null, answer_bytes, input_bytes);
  return similarity.ratio();
};

// var a = "안녕";
// var b = "안녕?";
// console.log(similarity_check(a, b));
